usaspending_api/etl/elasticsearch_loader_helpers/load_data.py

```diff
@@ -72,7 +72,7 @@ def streaming_post_to_es(
     try:
         if delete_before_index:
             value_list = [doc[delete_key] for doc in chunk]
-            delete_docs_by_unique_key(client, delete_key, value_list, job_name, index_name)
+            delete_docs_by_unique_key(client, delete_key, value_list, job_name, index_name, refresh_after=False)
```

Comment:
The new unit test for incremental load would indicate that this works. Meaning, you can delete a doc and re-add the updated version of it without doing a refresh in-between. That's the assumption I was going on, but merits more real-testing against larger datasets and incremental loads.

The thinking / motivation here is that if we are pushing batches of 4000 or 10000 docs to ES at a time, and having to do and covering A LOT of records (say 1M or 2M updates) that's hundreds of refresh calls during the processing, covering thousands of documents each time to refresh, which adds drag to the overall process. So if we can **defer** the refresh of each batch and do only a refresh one all batches are done (or when an error occurs), it would streamline the data ingest. 

That ^ is based on the assumption that you can _add_ a new doc the same as the one just deleted _without_ having to do a refresh in between. I'll test to be sure that's the case.

Reply:
Was your test cluster multi-node or a single node?

I definitely understand the performance hit and the annoyance of calling refresh. Depends if a delete is eventually consistent or consistent. It might be a price for horizontal scaling technology.

Reply:
Seems to have done okay with processing about 2M transaction updates from the `transaction_search` table to the transaction ETL (deletes before re-insert on transaction index) `elasticsearch_indexer_etl` job number `48`. Looks like the refreshes were held off, but ES still seems to trigger some kind of refresh as needed at intervals of 500k docs or so based on the stair-step of deleted docs during the run. 
![image](https://user-images.githubusercontent.com/14953413/106781914-9715c780-6617-11eb-9404-7a55b12b9adb.png)